BWA_mutect_twist/aggregate_variants_mutect.R

```diff
@@ -1,236 +1,175 @@
-    str_replace_all("cDnaChange", "cDNA_Change") %>% 
-    str_replace_all("codonChange", "Codon_Change") %>%
-    str_replace_all("otherTranscripts", "Other_Transcripts") %>% 
-    str_replace_all("gcContent", "gc_content") %>% 
-    str_replace_all("referenceContext", "ref_context") %>%
-    str_replace_all("^start$", "Start_Position") %>% 
-    str_replace_all("^end$", "End_Position") %>% 
-    str_replace_all("^AF$", "tumor_f") %>% 
-    str_replace_all("Oreganno", "OREGANNO")
-
-mutect_vcf_filter <- filter(mutect_vcf_bind, is_in(Variant_Classification, c("FRAME_SHIFT_DEL", "FRAME_SHIFT_INS", "MISSENSE", "NONSENSE", "SPLICE_SITE"))) %>% 
-    filter(nchar(Protein_Change) > 0)
-
-mutect_vcf_ncol <- str_split(mutect_vcf_filter$AD, ",") %>% map_int(length) %>% max
-
-Split_Columns <- function(column, nocolumns, name1, name2, make_numeric) {
-  object<-str_split_fixed(column, ",", nocolumns)  %>%
-    as_tibble
-    if (make_numeric == TRUE){
-      object <- mutate(object, across(everything(), as.numeric))  }
-  colnames(object) <- c(name1, str_c(name2, seq(1:(nocolumns- 1)))) 
-  object
-}
-mutect_vcf_ad <-Split_Columns(mutect_vcf_filter$AD, mutect_vcf_ncol, "t_ref_count", "t_alt_count_", TRUE)
-mutect_vcf_af <- Split_Columns(mutect_vcf_filter$tumor_f, mutect_vcf_ncol, "tumor_f", "", TRUE)
-mutect_vcf_f2r1 <- Split_Columns(mutect_vcf_filter$F2R1, mutect_vcf_ncol, "f2r1_reference", "f2r1_alternate", TRUE)
-mutect_vcf_f1r2 <- Split_Columns(mutect_vcf_filter$F1R2, mutect_vcf_ncol, "f1r2_reference", "f1r2_alternate", TRUE)
-mutect_vcf_mbq <- Split_Columns(mutect_vcf_filter$MBQ, mutect_vcf_ncol, "mbq_reference", "mbq_alternate", TRUE)
-mutect_vcf_mmq <- Split_Columns(mutect_vcf_filter$MMQ, mutect_vcf_ncol, "mmq_reference", "mmq_alternate", TRUE)
-mutect_vcf_mfrl <- Split_Columns(mutect_vcf_filter$MFRL, mutect_vcf_ncol, "mfrl_reference", "mfrl_alternate", TRUE)
-mutect_vcf_sb <- str_split_fixed(mutect_vcf_filter$AS_SB_TABLE, "\\|", mutect_vcf_ncol) %>% as_tibble 
+mutect_data_df <- map2(mutect_data_names, mutect_data, SetNames) %>% bind_rows %>% select(-DP)
+
+mutect_vcf_bind <- select(mutect_vcf_all, Sample:FILTER) %>% bind_cols(mutect_data_df) %>% bind_cols(mutect_info_df) 
+
+# Extract the maf data using the make_data function and bind the resulting data into one tibble
+list_of_mafs <- maf_files %>%
+  map(read.maf)%>%
+  map(make_data)%>%
+  map(as_tibble)%>%
+  set_names(sample_names)
+mutect_maf_all <- bind_rows(list_of_mafs, .id = "Sample")
+
+# Remove all mitochondrial genes
+maf_noMT <- subset(mutect_maf_all, mutect_maf_all$Chromosome != "MT")
+
+# IF a gene's Reference_Allele column contains an A, T, G, or C AND their Tumor_Seq_Allele2 column is blank ("-")
+  # THEN, subtract one from the Start_Position column
+TSA_empty <- filter(maf_noMT, Reference_Allele != "-" & Tumor_Seq_Allele2 == "-")
+Start_Position_Mutated <- TSA_empty %>%
+  mutate(Start_Position = as.integer(Start_Position))%>%
```

Comment:
Let's use `mutate(TSA_empty, Start_Position = as.integer(Start_Position))` instead of piping the data frame into the first function.